lib/dradis/plugins/template_service.rb

```diff
@@ -90,11 +83,55 @@ def template_source
 
       private
 
+      def apply_mapping(mapping)
+        # fetch mapping_fields through mapping or default mapping fields through plugin
+        mapping_fields =
+          if mapping && mapping.mapping_fields.any?
+            mapping.mapping_fields
+          else
+            plugin::Mapping.default_mapping[template]
+          end
+
+        mapping_fields.map do |field|
+          field_name = field.try(:destination_field) || field[0]
+          field_content = process_content(field.try(:content) || field[1])
+          "#[#{field_name.titleize}]#\n\n#{field_content}"
+        end
+      end
+
       # This method returns the default location in which plugins should look
       # for their templates.
       def default_templates_dir
         @default_templates_dir ||= begin
-          File.join(Configuration.paths_templates_plugins, @plugin::meta[:name].to_s)
+          File.join(Configuration.paths_templates_plugins, plugin::meta[:name].to_s)
+        end
+      end
+
+      def mapping
+        return nil unless project
+        rtp =
+          if project.report_template_properties
+            "rtp_#{project.report_template_properties_id}"
+          else
+            nil # allow for nil in CE
+          end
+
+        Mapping.includes(:mapping_fields).find_by(
+          component: plugin.name.demodulize.downcase,
```

Comment:
This is knowledge that looks like it's going to be needed elsewhere (i.e. how do we go from plugin to component). Wouldn't we be better off encapsulating it in the plugin itself?

Otherwise, we have to remember and keep track of this chain of methods all over the code base (i.e. across repos!).

Reply:
Added a component_name helper to each plugin so that we have a single source of truth for building/fetching mappings

Reply:
Now passing `component` to `mapping_service` instead